numba/targets/randomimpl.py

```diff
@@ -1145,26 +1145,60 @@ def zipf_impl(a):
     res = context.compile_internal(builder, zipf_impl, sig, args)
     return impl_ret_untracked(context, builder, sig.return_type, res)
 
-
-@lower("random.shuffle", types.Buffer)
-def shuffle_impl(context, builder, sig, args):
-    res = _shuffle_impl(context, builder, sig, args, random.randrange)
-    return impl_ret_untracked(context, builder, sig.return_type, res)
-
-@lower("np.random.shuffle", types.Buffer)
-def shuffle_impl(context, builder, sig, args):
-    res = _shuffle_impl(context, builder, sig, args, np.random.randint)
-    return impl_ret_untracked(context, builder, sig.return_type, res)
-
-def _shuffle_impl(context, builder, sig, args, _randrange):
-    def shuffle_impl(arr):
-        i = arr.shape[0] - 1
-        while i > 0:
-            j = _randrange(i + 1)
-            arr[i], arr[j] = arr[j], arr[i]
-            i -= 1
-
-    return context.compile_internal(builder, shuffle_impl, sig, args)
+def _shuffle_impl(arr, rng):
+    pass
+
+@overload(_shuffle_impl)
```

Comment:
@stuartarchibald I think `@overload` doesn't make sense here since the `rng` argument is a string. Also, `_shuffle_impl` isn't used anywhere.
Other than this, the patch looks good to me.

Reply:
Okay, removed those lines.  @stuartarchibald, I've also updated the docs, so this can be merged, I think.